go-controller/pkg/ovn/base_network_controller_secondary.go

```diff
@@ -836,6 +841,36 @@ func (bsnc *BaseSecondaryNetworkController) buildUDNEgressSNAT(localPodSubnets [
 	return snats, nil
 }
 
+func (bsnc *BaseSecondaryNetworkController) ensureDHCP(pod *corev1.Pod, podAnnotation *util.PodAnnotation, lsp *nbdb.LogicalSwitchPort) error {
+
+	// Configure DHCP only for kubevirt VMs layer2 primary udn with subnets
+	if !kubevirt.VirtualMachineOwnsPod(pod) ||
+		!util.IsNetworkSegmentationSupportEnabled() ||
+		!bsnc.IsPrimaryNetwork() ||
+		bsnc.TopologyType() != types.Layer2Topology ||
+		len(bsnc.Subnets()) == 0 {
+		return nil
+	}
+
+	opts := []kubevirt.DHCPConfigsOpt{}
+
+	ipv4DNSerer, ipv6DNServer, err := kubevirt.RetrieveDNSServiceClusterIPs(bsnc.watchFactory)
+	if err != nil {
+		return err
+	}
+
+	ipv4Gateway, _ := util.MatchFirstIPFamily(false /*ipv4*/, podAnnotation.Gateways)
+	if ipv4Gateway != nil {
+		opts = append(opts, kubevirt.WithIPv4Router(ipv4Gateway.String()))
+	}
+
+	//TODO classless routes
```

Comment:
I needed to ask maintainers about it, since current routes are redundant, the default route is enough, but looks like after services is implemented we may need them.

About implementation for IPv4 we have 

```
options : classless_static_route: optional string
              The DHCPv4 option code for this option is 121.

              This option can contain one or more static routes, each of
              which consists of a destination descriptor and the IP
              address of the router that should be used to reach that
              destination. Please see RFC 3442 for more details.

              Example: {30.0.0.0/24,10.0.0.10, 0.0.0.0/0,10.0.0.1}
```

I have to check what we do for IPv6.

Reply:
@maiqueb after some discussion with maintainers is ok to no implement classless routes, we may need to re-visit this for L2 secondary interfaces if ingress is implemented there using k8s services.